keras_nlp/tokenizers/byte_pair_tokenizer.py

```diff
@@ -51,9 +52,18 @@
 SPLIT_PATTERN_1 = SPLIT_PATTERN_1.replace(
     "{special_spaces}", SPECIAL_WHITESPACES
 )
+SPLIT_PATTERN_2 = rf"""[\s६{SPECIAL_WHITESPACES}]$"""
 
 
-SPLIT_PATTERN_2 = rf"""[\s६{SPECIAL_WHITESPACES}]$"""
+def create_alts_for_special_tokens(special_tokens):
+    alts = []
```

Comment:
maybe leave a comment to better describe this...

Create alternates for all special tokens that will be unsplitting during tokenization.